--- nss-pam-ldapd-0.7.0/nslcd.conf~	2009-08-31 23:46:06.000000000 +0200
+++ nss-pam-ldapd-0.7.0/nslcd.conf	2009-09-17 23:30:09.251930208 +0200
@@ -4,6 +4,10 @@
 # information in the directory.
 # See the manual page nslcd.conf(5) for more information.
 
+#threads 5
+uid nslcd
+gid nobody
+
 # The uri pointing to the LDAP server to use for name lookups.
 # Multiple entries may be specified. The address that is used
 # here should be resolvable without using LDAP (obviously).
