diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/alias.c nss-pam-ldapd-0.9.4/nslcd/alias.c
--- nss-pam-ldapd-0.9.4.org/nslcd/alias.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/alias.c	2025-11-24 19:07:39.200021750 +0100
@@ -72,8 +72,8 @@ static int mkfilter_alias_byname(const c
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    alias_filter, attmap_alias_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_alias_cn, safename, alias_filter);
 }
 
 void alias_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/ether.c nss-pam-ldapd-0.9.4/nslcd/ether.c
--- nss-pam-ldapd-0.9.4.org/nslcd/ether.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/ether.c	2025-11-24 19:07:39.200235492 +0100
@@ -74,8 +74,8 @@ static int mkfilter_ether_byname(const c
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    ether_filter, attmap_ether_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_ether_cn, safename, ether_filter);
 }
 
 static int mkfilter_ether_byether(const char *addrstr,
@@ -85,8 +85,8 @@ static int mkfilter_ether_byether(const
      representation (e.g. 1:0:e:...) and not with extra leading zeros
      (e.g. 01:00:0e:...) */
   /* there should be no characters that need escaping */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    ether_filter, attmap_ether_macAddress, addrstr);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_ether_macAddress, addrstr, ether_filter);
 }
 
 void ether_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/group.c nss-pam-ldapd-0.9.4/nslcd/group.c
--- nss-pam-ldapd-0.9.4.org/nslcd/group.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/group.c	2025-11-24 19:07:39.200431833 +0100
@@ -112,19 +112,21 @@ static int mkfilter_group_bygid(gid_t gi
   {
     /* the given gid is a BUILTIN gid, the SID prefix is not the domain SID */
     if ((gid >= min_builtin_rid) && (gid <= max_builtin_rid))
-      return mysnprintf(buffer, buflen, "(&%s(%s=%s\\%02x\\%02x\\%02x\\%02x))",
-                        group_filter, attmap_group_gidNumber, builtinSid,
+      return mysnprintf(buffer, buflen, "(&(%s=%s\\%02x\\%02x\\%02x\\%02x)%s)",
+                        attmap_group_gidNumber, builtinSid,
                         (int)(gid & 0xff), (int)((gid >> 8) & 0xff),
-                        (int)((gid >> 16) & 0xff), (int)((gid >> 24) & 0xff));
-    return mysnprintf(buffer, buflen, "(&%s(%s=%s\\%02x\\%02x\\%02x\\%02x))",
-                      group_filter, attmap_group_gidNumber, gidSid,
+                        (int)((gid >> 16) & 0xff), (int)((gid >> 24) & 0xff),
+			group_filter);
+    return mysnprintf(buffer, buflen, "(&(%s=%s\\%02x\\%02x\\%02x\\%02x)%s)",
+                      attmap_group_gidNumber, gidSid,
                       (int)(gid & 0xff), (int)((gid >> 8) & 0xff),
-                      (int)((gid >> 16) & 0xff), (int)((gid >> 24) & 0xff));
+                      (int)((gid >> 16) & 0xff), (int)((gid >> 24) & 0xff),
+		      group_filter);
   }
   else
   {
-    return mysnprintf(buffer, buflen, "(&%s(%s=%d))",
-                      group_filter, attmap_group_gidNumber, (int)gid);
+    return mysnprintf(buffer, buflen, "(&(%s=%d)%s)",
+                      attmap_group_gidNumber, (int)gid, group_filter);
   }
 }
 
@@ -146,8 +148,8 @@ static int mkfilter_group_bymember(MYLDA
   /* try to translate uid to DN */
   if ((strcasecmp(attmap_group_member, "\"\"") == 0) ||
       (uid2dn(session, uid, dn, sizeof(dn)) == NULL))
-    return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                      group_filter, attmap_group_memberUid, safeuid);
+    return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                      attmap_group_memberUid, safeuid, group_filter);
   /* escape DN */
   if (myldap_escape(dn, safedn, sizeof(safedn)))
   {
@@ -155,10 +157,9 @@ static int mkfilter_group_bymember(MYLDA
     return -1;
   }
   /* also lookup using user DN */
-  return mysnprintf(buffer, buflen, "(&%s(|(%s=%s)(%s=%s)))",
-                    group_filter,
+  return mysnprintf(buffer, buflen, "(&(|(%s=%s)(%s=%s))%s)",
                     attmap_group_memberUid, safeuid,
-                    attmap_group_member, safedn);
+                    attmap_group_member, safedn, group_filter);
 }
 
 static int mkfilter_group_bymemberdn(const char *dn,
@@ -172,9 +173,8 @@ static int mkfilter_group_bymemberdn(con
     return -1;
   }
   return mysnprintf(buffer, buflen,
-                    "(&%s(%s=%s))",
-                    group_filter,
-                    attmap_group_member, safedn);
+                    "(&(%s=%s)%s)",
+                    attmap_group_member, safedn, group_filter);
 }
 
 void group_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/host.c nss-pam-ldapd-0.9.4/nslcd/host.c
--- nss-pam-ldapd-0.9.4.org/nslcd/host.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/host.c	2025-11-24 19:07:39.200616818 +0100
@@ -74,8 +74,8 @@ static int mkfilter_host_byname(const ch
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    host_filter, attmap_host_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_host_cn, safename, host_filter);
 }
 
 static int mkfilter_host_byaddr(const char *addrstr,
@@ -89,8 +89,8 @@ static int mkfilter_host_byaddr(const ch
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    host_filter, attmap_host_ipHostNumber, safeaddr);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_host_ipHostNumber, safeaddr, host_filter);
 }
 
 void host_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/netgroup.c nss-pam-ldapd-0.9.4/nslcd/netgroup.c
--- nss-pam-ldapd-0.9.4.org/nslcd/netgroup.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/netgroup.c	2025-11-24 19:07:39.200758671 +0100
@@ -73,8 +73,8 @@ static int mkfilter_netgroup_byname(cons
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    netgroup_filter, attmap_netgroup_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_netgroup_cn, safename, netgroup_filter);
 }
 
 void netgroup_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/network.c nss-pam-ldapd-0.9.4/nslcd/network.c
--- nss-pam-ldapd-0.9.4.org/nslcd/network.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/network.c	2025-11-24 19:07:39.200912106 +0100
@@ -73,8 +73,8 @@ static int mkfilter_network_byname(const
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    network_filter, attmap_network_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_network_cn, safename, network_filter);
 }
 
 static int mkfilter_network_byaddr(const char *addrstr,
@@ -88,8 +88,8 @@ static int mkfilter_network_byaddr(const
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    network_filter, attmap_network_ipNetworkNumber, safeaddr);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_network_ipNetworkNumber, safeaddr, network_filter);
 }
 
 void network_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/passwd.c nss-pam-ldapd-0.9.4/nslcd/passwd.c
--- nss-pam-ldapd-0.9.4.org/nslcd/passwd.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/passwd.c	2025-11-24 19:07:39.201084832 +0100
@@ -95,8 +95,8 @@ static int mkfilter_passwd_byname(const
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    passwd_filter, attmap_passwd_uid, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_passwd_uid, safename, passwd_filter);
 }
 
 /* create a search filter for searching a passwd entry
@@ -105,15 +105,16 @@ static int mkfilter_passwd_byuid(uid_t u
 {
   if (uidSid != NULL)
   {
-    return mysnprintf(buffer, buflen, "(&%s(%s=%s\\%02x\\%02x\\%02x\\%02x))",
-                      passwd_filter, attmap_passwd_uidNumber, uidSid,
+    return mysnprintf(buffer, buflen, "(&(%s=%s\\%02x\\%02x\\%02x\\%02x)%s)",
+                      attmap_passwd_uidNumber, uidSid,
                       (int)(uid & 0xff), (int)((uid >> 8) & 0xff),
-                      (int)((uid >> 16) & 0xff), (int)((uid >> 24) & 0xff));
+                      (int)((uid >> 16) & 0xff), (int)((uid >> 24) & 0xff),
+		      passwd_filter);
   }
   else
   {
-    return mysnprintf(buffer, buflen, "(&%s(%s=%d))",
-                      passwd_filter, attmap_passwd_uidNumber, (int)uid);
+    return mysnprintf(buffer, buflen, "(&(%s=%d)%s)",
+                      attmap_passwd_uidNumber, (int)uid, passwd_filter);
   }
 }
 
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/protocol.c nss-pam-ldapd-0.9.4/nslcd/protocol.c
--- nss-pam-ldapd-0.9.4.org/nslcd/protocol.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/protocol.c	2025-11-24 19:07:39.201270849 +0100
@@ -73,8 +73,8 @@ static int mkfilter_protocol_byname(cons
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    protocol_filter, attmap_protocol_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_protocol_cn, safename, protocol_filter);
 }
 
 /* create a search filter for searching a protocol entry
@@ -82,8 +82,8 @@ static int mkfilter_protocol_byname(cons
 static int mkfilter_protocol_bynumber(int protocol,
                                       char *buffer, size_t buflen)
 {
-  return mysnprintf(buffer, buflen, "(&%s(%s=%d))",
-                    protocol_filter, attmap_protocol_ipProtocolNumber, protocol);
+  return mysnprintf(buffer, buflen, "(&(%s=%d)%s)",
+                    attmap_protocol_ipProtocolNumber, protocol, protocol_filter);
 }
 
 void protocol_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/rpc.c nss-pam-ldapd-0.9.4/nslcd/rpc.c
--- nss-pam-ldapd-0.9.4.org/nslcd/rpc.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/rpc.c	2025-11-24 19:07:39.201406831 +0100
@@ -74,14 +74,14 @@ static int mkfilter_rpc_byname(const cha
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    rpc_filter, attmap_rpc_cn, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_rpc_cn, safename, rpc_filter);
 }
 
 static int mkfilter_rpc_bynumber(int number, char *buffer, size_t buflen)
 {
-  return mysnprintf(buffer, buflen, "(&%s(%s=%d))",
-                    rpc_filter, attmap_rpc_oncRpcNumber, number);
+  return mysnprintf(buffer, buflen, "(&(%s=%d)%s)",
+                    attmap_rpc_oncRpcNumber, number, rpc_filter);
 }
 
 void rpc_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/service.c nss-pam-ldapd-0.9.4/nslcd/service.c
--- nss-pam-ldapd-0.9.4.org/nslcd/service.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/service.c	2025-11-24 19:07:39.201561957 +0100
@@ -83,13 +83,14 @@ static int mkfilter_service_byname(const
       log_log(LOG_ERR, "mkfilter_service_byname(): safeprotocol buffer too small");
       return -1;
     }
-    return mysnprintf(buffer, buflen, "(&%s(%s=%s)(%s=%s))",
-                      service_filter, attmap_service_cn, safename,
-                      attmap_service_ipServiceProtocol, safeprotocol);
+    return mysnprintf(buffer, buflen, "(&(%s=%s)(%s=%s)%s)",
+                      attmap_service_cn, safename,
+                      attmap_service_ipServiceProtocol, safeprotocol,
+		      service_filter);
   }
   else
-    return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                      service_filter, attmap_service_cn, safename);
+    return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                      attmap_service_cn, safename, service_filter);
 }
 
 static int mkfilter_service_bynumber(int number, const char *protocol,
@@ -103,13 +104,14 @@ static int mkfilter_service_bynumber(int
       log_log(LOG_ERR, "mkfilter_service_bynumber(): safeprotocol buffer too small");
       return -1;
     }
-    return mysnprintf(buffer, buflen, "(&%s(%s=%d)(%s=%s))",
-                      service_filter, attmap_service_ipServicePort, number,
-                      attmap_service_ipServiceProtocol, safeprotocol);
+    return mysnprintf(buffer, buflen, "(&(%s=%d)(%s=%s)%s)",
+                      attmap_service_ipServicePort, number,
+                      attmap_service_ipServiceProtocol, safeprotocol,
+		      service_filter);
   }
   else
-    return mysnprintf(buffer, buflen, "(&%s(%s=%d))",
-                      service_filter, attmap_service_ipServicePort, number);
+    return mysnprintf(buffer, buflen, "(&(%s=%d)%s)",
+                      attmap_service_ipServicePort, number, service_filter);
 }
 
 void service_init(void)
diff -urNp -x '*.orig' nss-pam-ldapd-0.9.4.org/nslcd/shadow.c nss-pam-ldapd-0.9.4/nslcd/shadow.c
--- nss-pam-ldapd-0.9.4.org/nslcd/shadow.c	2014-06-06 17:28:40.000000000 +0200
+++ nss-pam-ldapd-0.9.4/nslcd/shadow.c	2025-11-24 19:07:39.201718743 +0100
@@ -80,8 +80,8 @@ static int mkfilter_shadow_byname(const
     return -1;
   }
   /* build filter */
-  return mysnprintf(buffer, buflen, "(&%s(%s=%s))",
-                    shadow_filter, attmap_shadow_uid, safename);
+  return mysnprintf(buffer, buflen, "(&(%s=%s)%s)",
+                    attmap_shadow_uid, safename, shadow_filter);
 }
 
 void shadow_init(void)
